crates/services/txpool/src/test_helpers.rs

```diff
@@ -49,7 +49,7 @@ pub(crate) fn add_coin_to_state(input: Input, mock_db: Option<&MockDb>) -> (Coin
     let coin = CompressedCoin {
         owner: *input.input_owner().unwrap(),
         amount: TEST_COIN_AMOUNT,
-        asset_id: *input.asset_id().unwrap(),
+        asset_id: *input.asset_id(&AssetId::BASE).unwrap(),
```

Comment:
This is about an upstream dep, so not something you can fix, but I don't love this syntax. There is another place where we pass the `AssetId::BASE` into a getter. I've looked at the source code for these methods and it seems like we return the `base_asset_id` for certain variants... so a getter that is a passthrough.

I think it would be better if those variants could know what their base asset is.

Again, not something you need to fix.

Reply:
I tend to agree that passing a value to a getter is a bit unorthodox, and it might be more desirable for the underlying struct to know what the base asset is. The challenge here specifically is that `Input`s, specifically `Message`s and `MessageCoin`s  are defined by a specification, and this includes their serialization and deserialization. The data is fed in via messages, and the message specification _does not_ include the base asset id, so we would not be able to deserialize a message if we put the base asset on the struct. Similarly, we would have to include workaround logic to handle serialization.

Reply:
Yeah. I figured it was something along those lines. It's not the worst thing, just a smell. I'm not gonna worry about it too much.